recipes/tcl/8.6.9/conanfile.py

```diff
@@ -0,0 +1,220 @@
+from conans.errors import ConanExceptionInUserConanfileMethod
+from conans.util.env_reader import get_env
+import os
+import shutil
+import tempfile
+
+
+class TclConan(ConanFile):
+    name = "tcl"
+    description = "Tcl is a very powerful but easy to learn dynamic programming language."
+    topics = ("conan", "tcl", "scripting", "programming")
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://tcl.tk"
+    license = "TCL"
+    settings = "os", "compiler", "build_type", "arch"
+    options = {
+        "fPIC": [True, False],
+        "shared": [True, False]
+    }
+    default_options = {
+        "fPIC": True,
+        "shared": False,
+    }
+    _source_subfolder = "sources"
+    requires = ("zlib/1.2.11")
+
+    @property
+    def _is_mingw_windows(self):
+        return self.settings.os == "Windows" and self.settings.compiler == "gcc"
+
+    @property
+    def _install_folder(self):
+        # conan-center-index forbids having share/man folders at the package root
+        # it also advises to put everything into a bin folder
+        # until there is a better way, let's go for the bin/bin weirdness
+        return os.path.join(self.package_folder, "bin").replace("\\", "/")
+
+    def configure(self):
+        if self.settings.compiler != "Visual Studio":
+            del self.settings.compiler.libcxx
+            del self.settings.compiler.cppstd
+
+    def build_requirements(self):
+        if self._is_mingw_windows:
+            self.build_requires("msys2/20161025")
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        extracted_dir = self.name + self.version
+        os.rename(extracted_dir, self._source_subfolder)
+
+    def _fix_sources(self):
+        unix_config_dir = self._get_configure_dir("unix")
+        # When disabling 64-bit support (in 32-bit), this test must be 0 in order to use "long long" for 64-bit ints
+        # (${tcl_type_64bit} can be either "__int64" or "long long")
+        tools.replace_in_file(os.path.join(unix_config_dir, "configure"),
+                              "(sizeof(${tcl_type_64bit})==sizeof(long))",
+                              "(sizeof(${tcl_type_64bit})!=sizeof(long))")
+
+        unix_makefile_in = os.path.join(unix_config_dir, "Makefile.in")
+        # Avoid building internal libraries as shared libraries
+        tools.replace_in_file(unix_makefile_in, "--enable-shared --enable-threads", "--enable-threads")
+        # Avoid clearing CFLAGS and LDFLAGS in the makefile
+        tools.replace_in_file(unix_makefile_in, "\nCFLAGS\t", "\n#CFLAGS\t")
+        tools.replace_in_file(unix_makefile_in, "\nLDFLAGS\t", "\n#LDFLAGS\t")
+        # Use CFLAGS and CPPFLAGS as argument to CC
+        tools.replace_in_file(unix_makefile_in, "${CFLAGS}", "${CFLAGS} ${CPPFLAGS}")
+
+    def config_options(self):
+        if self.settings.compiler == "Visual Studio" or self.options.shared:
```

Comment:
```suggestion
        if self.settings.os == "Windows":
```
Try to follow what most of recipes are using. It can make difference when building with `-o *:fPIC=True/False`